Themis/ViewModels/Assessment/AssessmentViewModel.swift

```diff
@@ -1,72 +1,20 @@
 import Foundation
 import SwiftUI
+import Combine
 
 class AssessmentViewModel: ObservableObject {
-    @Published var fileTree: [Node] = []
-    @Published var openFiles: [Node] = []
-    @Published var selectedFile: Node?
-    @Published var editorFontSize = CGFloat(14) // Default font size
-
-    init(files: [String: FileType]) {
-        let node = ArtemisAPI.initFileTreeStructure(files: files)
-        self.fileTree = node.children ?? []
-    }
-
-    func openFile(file: Node) {
-        if !openFiles.contains(where: { $0.path == file.path }) {
-            openFiles.append(file)
-            file.fetchCode()
+    @Published var submission: SubmissionForAssessment?
+    @Published var foundSubmission: Bool = false
+
+    @MainActor
+    func initRandomSubmission(exerciseId: Int) async {
+        do {
+            self.submission = try await ArtemisAPI.getRandomSubmissionForAssessment(exerciseId: 5284)
```

Comment:
`exerciseId` should not be hardcoded here
```suggestion
            self.submission = try await ArtemisAPI.getRandomSubmissionForAssessment(exerciseId: exerciseId)
```

Reply:
Good call